### DISCUSSION TITLE
Git in xterm.js

### DISCUSSION THREAD

**Discussion by iliakan** [sourcelair/xterm.js#4378] [Q&A] [answered]
I see some sites like repl.it or codeschool with a "Web Terminal" allowing to call git in xterm, and this git is totally like a "real" CLI.

How is it done technically?

Is that git a part of xterm.js? or is there a real backend container running somewhere, and connected with?

P.S. If so, any ideas about container management for an open source online tutorial? ;)

**Comment by Tyriar** [comment, marked as answer]
There is normally a backend that xterm.js (the frontend) connects to. You can see a simple example of this in the demo which connects to a local server: https://github.com/xtermjs/xterm.js/wiki/Contributing#running-the-demo

**Comment by iliakan** [comment]
That's what I thought :(

Any idea where to find containers for a free open source project? For people to experiment in.

**Reply by jerch** [reply]
Well for free containers - there are a ton to choose from on docker hub.

The main issue is more where to get computing resources for free, as these containers have to run on some (virtual) hardware somewhere. For a very small solution, that does not scale by any means, you might have luck with limited starter packages offered by some hosters (years ago heroku had a free starter plan, no clue if they still offer that).

For anything more reliable you'll have to buy the server resources yourself.
